src/components/ReportsPage/HeatmapPage/HeatmapTable.tsx

```diff
@@ -0,0 +1,226 @@
+}
+
+export const HeatmapTable: React.FC<HeatmapTableProps> = ({
+  allDomains,
+  allTeams,
+  catalog,
+  domainAncestryMap,
+  domainHierarchy,
+  entitiesByTag,
+  filters,
+  ladder,
+  scorecard,
+  scores,
+  setFilters,
+  teamsByEntity,
+  teamHierarchy,
+}) => {
+  const colorStyles = useColorCellStyles();
+
+  const {
+    mappedDomainAncestryMap,
+    mappedDomainHierarchies,
+    mappedScorecard,
+    mappedScores,
+  } = useMemo(() => {
+    const mappedScorecard = convertToBirdsEyeScorecard(scorecard, ladder);
+    const mappedScores = convertToBirdsEyeScores(scores, catalog);
+    const mappedDomainHierarchies =
+      convertToBirdsEyeDomainHierarchy(domainHierarchy);
+    const mappedDomainAncestryMapKeys = mapKeys(domainAncestryMap, key =>
+      key.toString(),
+    );
+    const mappedDomainAncestryMap = mapValues(
+      mappedDomainAncestryMapKeys,
+      values => values.map(value => value.toString()),
+    );
+
+    return {
+      mappedScorecard,
+      mappedScores,
+      mappedDomainHierarchies,
+      mappedDomainAncestryMap,
+    };
+  }, [catalog, domainAncestryMap, domainHierarchy, ladder, scorecard, scores]);
+
+  const {
+    tableData,
+    setDataFilters,
+    groupByOptions,
+    setReportType,
+    filtersConfig,
+    setHideTeamsWithoutEntities,
+    setGroupBy,
+    setUseHierarchy,
+    showHierarchy,
+    groupBy,
+    shouldShowReportType,
+  } = useCortexBirdseye({
+    allDomains,
+    allTeams,
+    domainAncestryMap: mappedDomainAncestryMap,
+    domainHierarchy: mappedDomainHierarchies,
+
+    filters,
+    scorecard: mappedScorecard,
+    scores: mappedScores,
+    setFilters,
+    teamHierarchy,
+    teamsByEntity,
+  });
+
+  const scorecardServiceDetailsRef = useRouteRef(
+    scorecardServiceDetailsRouteRef,
+  );
+
+  const cellColorClassNameHandler = useCallback(
+    (points: number) => {
+      return `${colorStyles.root} ${
+        colorStyles[getCellColorBackground(points)]
+      }`;
+    },
+    [colorStyles],
+  );
+
+  const scoreColorClassNameHandler = useCallback(
+    (points: number) => {
+      return `${colorStyles.root} ${
+        colorStyles[getScoreColorClassName(points)]
+      }`;
+    },
+    [colorStyles],
+  );
+
+  const scorecardEntityUrlHandler = useCallback(
+    (
+      scorecardEntity: Parameters<
+        BirdsEyeReportTableProps['getScorecardEntityUrl']
+      >[0],
+    ) => {
+      const entityId = Number.parseInt(scorecardEntity.entityId);
+      const entity = catalog.find(entity => entityId === entity.id);
+      const codeTag = entity?.codeTag;
+      if (!codeTag) {
+        return '';
+      }
+
+      const componentRef = entityComponentRef(entitiesByTag, codeTag);
+
+      const entityName = parseEntityRef(
+        componentRef,
+        defaultComponentRefContext,
+      );
+
+      const entityUrl = scorecardServiceDetailsRef({
+        scorecardId: `${mappedScorecard.id}`,
+        ...entityName,
+      });
+
+      return entityUrl;
+    },
```

Comment:
We could also probably adjust birds eye to pass back the code tag and we can avoid this. Would that work?

Reply:
I'm not sure what you would like to avoid here. I'm currently getting the `codeTag` from the catalog response and I'm using it on another place too, so I will still need to send the catalog request.

Reply:
It's fine for now.
I was meaning that the birdseye report could pass back the entire entity and not just the ID and then you wouldn't have to look up the codeTag